source/decode.c

```diff
@@ -497,17 +616,22 @@ static int s_linestate_response(struct aws_http_decoder *decoder, struct aws_byt
 
     struct aws_byte_cursor version_expected = aws_http_version_to_str(AWS_HTTP_VERSION_1_1);
     if (!aws_byte_cursor_eq(&version, &version_expected)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_HTTP_STREAM, "id=%p: Incoming response uses unsupported HTTP version.", decoder->logging_id);
         return aws_raise_error(AWS_ERROR_HTTP_PARSE);
     }
 
-    int64_t code_val;
-    int err = s_read_int64(code, &code_val);
+    size_t code_val;
+    err = s_read_size(code, &code_val, 10);
```

Comment:
why 10?

Reply:
basis 10
I thought this was your interview question

Reply:
renamed function so this isn't confusing